client/my-sites/site-indicator/site-indicator.jsx

```diff
@@ -13,90 +15,115 @@ import Animate from 'components/animate';
 import ProgressIndicator from 'components/progress-indicator';
 import DisconnectJetpackButton from 'my-sites/plugins/disconnect-jetpack/disconnect-jetpack-button';
 import analytics from 'lib/analytics';
+import QuerySiteConnectionStatus from 'components/data/query-site-connection-status';
+import QuerySiteUpdates from 'components/data/query-site-updates';
 import { userCan } from 'lib/site/utils';
+import { isJetpackSite } from 'state/sites/selectors';
+import { getUpdatesBySiteId } from 'state/sites/updates/selectors';
+import { updateWordPress } from 'state/sites/updates/actions';
+import {
+	getSiteConnectionStatus,
+	getSiteCoreUpdateStatus,
+	isRequestingSiteConnectionStatus
+} from 'state/selectors';
+
+class SiteIndicator extends Component {
+	static propTypes = {
+		site: PropTypes.object,
```

Comment:
It is a connected component, but: we need the `site` object for accessing several things within the component (like `.options` for example), and it will not always use the selected ID, but rather gets the `site` as a prop from the parent component. This is expected, of course, because when we display the `SiteSelector`, we display an indicator for each site that needs it.